packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+| Bulk    | Boolean       | `FALSE` | If true, activates the bulk mode (multi-select)    |
+| Columns | Array[Object] | `[]`    | Holds the definitions for each column in the table |
+| Create  | Boolean       | `FALSE` | If true, activates the creation of new items       |
+| Delete  | Boolean       | `FALSE` | If true, activates the deletion of items           |
+| Update  | Boolean       | `FALSE` | If true, activates the edit mode                   |
+
+### Triggers
+
+Custom actions are present in most tables, they are an important aspect of all applications. The nature of the actions depends on the entity they are applied to, hence, an abstract way of defining custom actions is required.
+
+| Attribute | Type     | default      | Description                                                 |
+| --------- | -------- | ------------ | ----------------------------------------------------------- |
+| id        | String   | `''`         | The id of the action, usually the name.                     |
+| callback  | Function | `noop`       | The function to be called when the action button is clicked |
+| trigger   | ANY      | `<Button />` | The element that will trigger the action                    |
+
+### Data
+
+The data passed to the table for rendering purposes.
+
+| Name | Type          | default | Description                                   |
+| ---- | ------------- | ------- | --------------------------------------------- |
+| data | Array[Object] | `[]`    | the tabular data to be displayed by the table |
+
+### Columns
+
+The definition of columns depends on the data, we must make sure we have a well-defined set of data types to allow display and mutation for each one.
+
+| Attribute  | Type                                                                   | default    | Description                                              |
+| ---------- | ---------------------------------------------------------------------- | ---------- | -------------------------------------------------------- |
+| accessor   | String                                                                 | `null`     | The name of the entity attribute that maps to the column |
+| context    | Object                                                                 | `{}`       | the context to be passed down to the custom component    |
+| isEditable | Boolean                                                                | `FALSE`    | Specifies if the cells are editable                      |
+| label      | String                                                                 | `''`       | The text to display as column header                     |
+| render     | JSX                                                                    | `null`     | A custom component to render                             |
+| style      | Object                                                                 | `{}`       | Custom style for the rendered component                  |
```

Comment:
I worry about exposing either `render` or `style` props on these. In theory, Cascara should be providing all of the display options needed for data. Essentially, if there is need for a new display style for something in a List, Table, Card... we would want to add that to Cascara and control that by the `type` attribute.

Opening up these two is probably too much and allows for the design system opinions to be bypassed. This is a slight shift in thinking from a UI library where we actually want to expose as many permutations for layout as possible. With Cascara we might be able to do these sorts of things with atomic pieces... but we might not be exporting all of those as public members. Cascara is the functional design system. So it has a lot more opinion added to it in terms of how things should work. Theoretically, that makes the APIs more simple that we expose... but the guts can be more complex than a UI library because of it.

- [ ] Discuss not using a render prop or exposing style

Reply:
I guess if there was a future scenario where we wanted composability in this pattern, we could include render... but I don't think we should call it `render` since there is the render prop pattern in React and people may confuse it with that. If we do happen to include this flexibility at some point... maybe we call this `component` instead. and it would override what is going on with `type`.

If we did that, I think it would also be good if we do not actually expose the `component` prop as part of the docs. Maybe it is hidden just in case. Still I think that might not be an MVP thing.

Reply:
I agree with you, that was an aspect I didn't like too much because of the related `context` thingy there. Just wanted to run this by you to hear your thoughts.

Also, I failed to mention that such extensibility option would be an enhancement. Just in case we haven't covered all actual scenarios. We could also create internal components to fill the gaps as you mention.

I will mmake the corrections to the wording here, thx!

Reply:
As agreed, the pattern we would follow is not the best and presents us with some risks. Removed `render` and `style` props.